digitaltwins/samples/device/javascript/pnpSampleDevice.js

```diff
@@ -0,0 +1,194 @@
+    console.log(request.payload);
+  };
+};
+
+const helperCreateReportedPropertiesPatch = (propertiesToReport, componentName, actualResponse) => {
+  const componentPart = componentPrefix + componentName;
+  const patch = { [componentPart]: {} };
+
+  Object.keys(propertiesToReport).forEach((propertyName) => {
+    const propertyContent = { value: propertiesToReport[propertyName] };
+    if (actualResponse) {
+      propertyContent.ac = actualResponse.code;
+      propertyContent.ad = actualResponse.description;
+      propertyContent.av = actualResponse.version;
+    }
+    patch[componentPart][propertyName] = propertyContent;
+  });
+  console.log('The following properties will be updated for component:' + componentName);
+  console.log(patch);
+  return patch;
+};
+
+const updateComponentReportedProperties = (deviceTwin, patch, componentName) => {
+  deviceTwin.properties.reported.update(patch, function (err) {
+    if (err) throw err;
+    console.log('Properties have been reported for component:' + componentName);
+  });
+};
+
+const helperAttachHandlerForDesiredPropertyPatches = (deviceTwin, componentName) => {
+  const prefixAndComponentName = componentPrefix + componentName;
+  const versionProperty = '$version';
+
+  deviceTwin.on('properties.desired.' + prefixAndComponentName, (delta) => {
+    Object.entries(delta).forEach(([propertyName, propertyValue]) => {
+      propertyUpdateHandler(deviceTwin, componentName, propertyName, null, propertyValue.value, deviceTwin.properties.desired[versionProperty]);
+    });
+  });
+};
+
+
+const sendTelemetryAtInterval = async (deviceClient, componentName, secs) => {
+  let index = 0;
+  const interval = parseFloat(secs)*1000;
+  setInterval( async () => {
+    console.log('Sending telemetry message %d...', index);
+    const data = JSON.stringify({ temp: 1 + (Math.random() * 90), humidity: 1 + (Math.random() * 99) });
+    const pnpMsg = new Message(data);
+    pnpMsg.properties.add(messageSubjectProperty, componentName);
+    pnpMsg.contentType = 'application/json';
+    pnpMsg.contentEncoding = 'utf-8';
+    await deviceClient.sendEvent(pnpMsg);
+    index += 1;
+  }, interval);
+};
+
+async function main() {
+  // fromConnectionString must specify a transport, coming from any transport package.
+  const client = Client.fromConnectionString(deviceConnectionString, Protocol);
+  console.log('Connecting using connection string ' + deviceConnectionString);
+  let resultTwin;
+  try {
+    // Add the modelId here
+    await client.setOptions({ modelId: modelId });
+    await client.open();
+    console.log('Enabling the commands on the client');
+    client.onDeviceMethod(commandNameBlink, commandHandler);
+    client.onDeviceMethod(commandNameTurnOn, commandHandler);
+    client.onDeviceMethod(commandNameTurnOff, commandHandler);
+    // Is there no way to register for an unknown method ?
+
+    // Send Telemetry every 5.5 secs
+    await sendTelemetryAtInterval(client, sensorComponentName, 5.5);
```

Comment:
>await sendTelemetryAtInterval(client, sensorComponentName, 5.5); [](start = 4, length = 64)

I'm thinking this is not good.  Essentially when you are saying await, you are saying wait here unitl setTelementryAtInterval resolves.  But I don't think you wait to wait here.  You want to go on immediately and execute the following code.